purchase_blanket_order/models/purchase_order.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2018 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models, _
+from datetime import date, timedelta
+from odoo.exceptions import ValidationError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    blanket_order_id = fields.Many2one(
+        'purchase.blanket.order', string='Origin blanket order',
+        related='order_line.blanket_order_line.order_id',
+        readonly=True)
+
+    @api.constrains('partner_id')
+    def check_partner_id(self):
+        if self.partner_id:
```

Comment:
@ageficent No need to check `if self.partner_id:` because In Odoo base `partner_id` is required